platform/mellanox/get_component_versions/get_component_versions.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from fwutil.lib import ComponentStatusProvider
+from tabulate import tabulate
+import os
+import subprocess
+import re
+
+COMPONENT_VERSIONS_FILE = "/etc/mlnx/component-versions"
+HEADERS = ["COMPONENT", "COMPILATION", "ACTUAL"]
+COMMANDS_FOR_ACTUAL = {
+    "MFT": ["dpkg -l | grep -e 'mft '", "mft *([0-9.]*)"],
+    "HW-MGMT": ["dpkg -l | grep hw", ".*1\\.mlnx\\.([0-9.]*)"],
+    "SDK": ["docker exec -it syncd bash -c 'dpkg -l | grep sdk'", ".*1\\.mlnx\\.([0-9.]*)"],
+    "SAI": ["docker exec -it syncd bash -c 'dpkg -l | grep mlnx-sai'", ".*1\\.mlnx\\.([A-Za-z0-9.]*)"],
+    "FW": ["mlxfwmanager --query | grep -e 'FW *[0-9.]*'", "FW * [0-9]{2}\\.([0-9.]*)"],
+    "Kernel": ["uname -r", "([0-9][0-9.-]*)-.*"]
+}
+UNAVAILABLE_PLATFORM_VERSIONS = {
+    "SSD": "N/A",
+    "BIOS": "N/A",
+    "CPLD": "N/A"
+}
+
+UNAVAILABLE_COMPILED_VERSIONS = {
+    "SDK": "N/A",
+    "FW": "N/A",
+    "SAI": "N/A",
+    "HW-MGMT": "N/A",
+    "MFT": "N/A",
+    "Kernel": "N/A"
+}
+
+
+def parse_compiled_components_file():
+    compiled_versions = {}
+
+    if not os.path.exists(COMPONENT_VERSIONS_FILE):
+        return UNAVAILABLE_COMPILED_VERSIONS
+
+    with open(COMPONENT_VERSIONS_FILE, 'r') as component_versions:
+        for line in component_versions.readlines():
+            comp, version = line.split()
+            compiled_versions[comp] = version
+
+    if len(compiled_versions.keys()) < len(UNAVAILABLE_COMPILED_VERSIONS.keys()):
+        for component in UNAVAILABLE_COMPILED_VERSIONS.keys():
+            if not compiled_versions.get(component):
+                compiled_versions[component] = "N/A"
+
+    return compiled_versions
+
+
+def get_platform_component_versions():
+    csp = ComponentStatusProvider()
+    version_table = csp.get_status()
```

Comment:
@skr31 why do you parse the string output? Suggest using `PlatformDataProvider` instead.
**Note:** Mellanox doesn't have modular chassis.